### PR TITLE
SC-1723: initial redux setup

### PR DESCRIPTION
**Pull request by samcartzach**
This is the first PR for real redux implementation. It includes:
* Sidebar active tab
* Canvas widgets
* Sidebar widgets
* WidgetList container

- basic actions setup
- basic reducer setup
- fix init state 
- minor cleanup of old code
- inclusion of actionType constants
- removes "example actions" and "example reducer"

### CODE REVIEW

File: src/reducers/canvasReducer.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import {
+  ADD_RENDERED_WIDGET,
+  DELETE_RENDERED_WIDGET,
+  UPDATE_RENDERED_WIDGET,
+  DUPLICATE_RENDERED_WIDGET,
+} from '../config/actionTypes';
+
+export default (state = {}, action) => {
+  switch (action.type) {
```
Comment (samcartalex) [inline src/reducers/canvasReducer.js]: not a huge deal, but you might want to 

`const { type } = action.payload;`

File: src/reducers/widgetListReducer.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  DISPLAY_WIDGETS,
+  FILTER_WIDGETS,
+} from '../config/actionTypes';
+
+export default (state = {}, action) => {
+  switch (action.type) {
+    case DISPLAY_WIDGETS:
+      return state.widgetList.filter(widget => widget.type === action.payload.type);
+    case FILTER_WIDGETS:
+      return state.widgetList.filter(widget => widget.category === action.payload.category
```
Comment (samcartalex) [inline src/reducers/widgetListReducer.js]: again, this is formatting preference, i think

```(widget =>(
     widget.category === action.payload.category
     && widget.type === action.payload.type
));
```

is easier to read but we could fight about that in standards

File: src/containers/WidgetListContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    super(props);
+    const { dispatch } = this.props;
+    this.displayWidgets = bindActionCreators(displayWidgets, dispatch);
+  }
+
+  render() {
+    const { widgetList } = this.props;
+    let sections = null;
+    if (widgetList) {
+      const widgetSections = widgetList.map(section => {
```
Comment (samcartalex) [inline src/containers/WidgetListContainer.jsx]: i think the standard is that any time you have a => and you're not all on the same line, you need ()

`widgetList.map((section) => {`

**Comment by samcartzach** [comment]
@samcartalex Updates made per request

**Pull request merged**